### PR TITLE
fix addressing image block overlay layout shift problems

### PR DESCRIPTION
**Pull request by myjeong19**
# Description of Changes

Here is reference: #25

---

## Review point

- Tooltip visibility when the scale input is focused.
- Tooltip styling.

## To reproduce

- npm run story:start
- click 'your created block' on the Storybook left side panel

## Review Guide

Reviews are conducted based on priority levels, such as p0, p1, p2, p3, p4, and p5.  
p0 ~ p2: If the author decides not to reflect a review for p0 to p2, it signals that a proper discussion with the reviewer is  
necessary. It is expected that the review will be resolved either through incorporating the feedback or through further discussion.  
p3: indicates that the reviewer has identified a significant issue, but either lacks a clear solution or the comment lacks sufficient context. Further explanation or additional discussion on the reviewer's concerns is needed.
p4, p5: p4 and p5 suggest low priority, and if the author does not deem them important, these comments can be disregarded.

### CODE REVIEW

File: packages/react-notion-custom/src/lib/index.css [hunk 1]
```diff
@@ -1317,3 +1025,282 @@ input::-webkit-inner-spin-button {
...
+}
+
+.notion-image-viewer-tools .notion-image-viewer-scaler input:focus {
+  color: white;
+  outline: 1px solid #2d75a0;
+  width: 28px;
+}
+/* Chrome, Safari, Edge, Opera */
+input::-webkit-outer-spin-button,
+input::-webkit-inner-spin-button {
```
Comment (Moon-DaeSeung) [inline packages/react-notion-custom/src/lib/index.css:1175]: p1: you should add .notion-image before input.

It affect all input element

File: packages/react-notion-custom/src/lib/index.css [hunk 2]
```diff
@@ -1317,3 +1025,282 @@ input::-webkit-inner-spin-button {
...
+  -webkit-appearance: none;
+  appearance: none;
+  margin: 0;
+}
+/* Firefox */
+input[type="number"] {
+  -moz-appearance: textfield;
```
Comment (Moon-DaeSeung) [inline packages/react-notion-custom/src/lib/index.css:1182]: p1: you should add .notion-image before input.

It affect all input element

File: packages/react-notion-custom/src/lib/components/image/image-viewer.tsx [hunk 3]
```diff
@@ -63,11 +62,14 @@ const ImageViewer: React.FC<ImageViewerProps> = ({
 
   useEffect(() => {
     if (!isOpened) {
+      document.body.style.overflow = "visible";
```
Comment (Moon-DaeSeung) [inline packages/react-notion-custom/src/lib/components/image/image-viewer.tsx:65]: p1: It cause layout shift (you know scroll bar has its own width)

File: packages/react-notion-custom/src/lib/components/image/image-viewer.tsx [hunk 4]
```diff
@@ -135,18 +137,18 @@ const ImageViewer: React.FC<ImageViewerProps> = ({
...
               ref={imageRef}
-              className={`notion-image-viewer-container-image ${isCursorVisible ? "notion-visible-cursor" : "notion-hide-cursor"}`}
+              className={`notion-image-viewer-container-image`}
               src={urls[currentImageIndex]}
               alt="posting image"
               style={{
                 transform: `scale(${scale})`,
                 transformOrigin: `${scaleOriginX * 100}% ${scaleOriginY * 100}%`,
-                cursor: getCursorStyle(scale),
+                cursor: isCursorVisible ? getCursorStyle(scale) : "none",
```
Comment (Moon-DaeSeung) [inline packages/react-notion-custom/src/lib/components/image/image-viewer.tsx:146]: p1: cursor does not disappear on overlay (not image). you should fix it

File: packages/react-notion-custom/src/lib/components/image/image-viewer.tsx (hunk 3 above)
Comment (Moon-DaeSeung) [reply packages/react-notion-custom/src/lib/components/image/image-viewer.tsx:65]: p1: you'd better research other library
but don't use it. just copy code

File: packages/core/src/lib/components/image/hooks/image-viewer/use-cursor-visibility.tsx [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { useState, useRef, useCallback } from "react";
 
-export const useCursorVisibility = (isScaleFocus: boolean) => {
+export const useCursorVisibility = () => {
   const [isCursorVisible, setIsisCursorVisible] = useState(true);
```
Comment (Moon-DaeSeung) [inline packages/core/src/lib/components/image/hooks/image-viewer/use-cursor-visibility.tsx:4]: p1: setIsis -> typo error

File: packages/core/src/lib/components/image/image-viewer.tsx [hunk 6]
```diff
@@ -61,6 +60,18 @@ const ImageViewer: React.FC<ImageViewerProps> = ({
     }
   }, [isOpened, currentImageIndex, setScale, setDisplayScale]);
 
+  useEffect(() => {
+    const notion = document.querySelector(".notion") as HTMLElement;
```
[the remark this answers is not in the document]
Comment (myjeong19) [reply packages/core/src/lib/components/image/image-viewer.tsx:64]: Is that still the case even when the image viewer is not open?

[the remark this answers is not in the document]
Comment (Moon-DaeSeung) [reply packages/core/src/lib/components/image/image-viewer.tsx:64]: umm.. scrolling element is not .notion

it depends on user's layout

**Comment by Moon-DaeSeung** [comment]
p3: you should give more detail info on pr title

**Comment by myjeong19** [comment]
I used react-remove-scroll-bar and react-style-singleton as references to resolve the layout shift issues.

File: packages/core/src/lib/components/image/image-viewer.tsx [hunk 7]
```diff
@@ -106,6 +106,26 @@ const ImageViewer: React.FC<ImageViewerProps> = ({
     [urls, setCurrentImageIndex, setIsOpened],
   );
 
+  useEffect(() => {
+    const styleElement = document.createElement("style");
+
+    if (isOpened) {
+      document.body.setAttribute("data-scroll-locked", "true");
+      const gap = getGapWidth();
```
Comment (Moon-DaeSeung) [inline packages/core/src/lib/components/image/image-viewer.tsx:114]: p2: you should capsulize this hook.

I suggest use-prevent-scroll

Comment (myjeong19) [reply packages/core/src/lib/components/image/image-viewer.tsx:114]: I checked this review.

**Review by Moon-DaeSeung — APPROVED** [review]
GOOOD

**Pull request merged**